src/main/java/ui/components/pickers/RepositoryPickerDialog.java

```diff
@@ -136,18 +165,76 @@ private void createUserInputTextField() {
         });
     }
 
+    /**
+     * Prompts the user for a repository id then return it only if it is a valid github repo. This method also informs
+     * the user if the repository that the user is trying to add is not valid.
+     */
+    private CompletableFuture<Boolean> addRepository() {
+        TextInputDialog dialog = new TextInputDialog();
+        dialog.setHeaderText("Add a new repository");
+        dialog.getEditor().textProperty().addListener((observable, oldValue, newValue) -> {
+            String repositoryId = Utility.removeAllWhitespace(newValue);
+            if (!repositoryId.equals(newValue)) {
+                dialog.getEditor().setText(repositoryId);
+                return;
+            }
+        });
+        Optional<String> userInput = dialog.showAndWait();
+        if (!userInput.isPresent()) {
+            return Futures.unit(false);
```

Comment:
We can use `CompletableFuture.completedFuture` (from the standard library) here. This `unit` method does the same and is a bit of a legacy thing.